src/Hl7.Fhir.ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -366,10 +366,10 @@ private IEnumerable<TypedElementOnSourceNode> enumerateElements(Dictionary<strin
                 childSet = parent.Children();
             else
             {
-                var hit = tryGetBySuffixedName(dis, name, out var info);
-                childSet = hit && info.IsChoiceElement ?
-                    parent.Children(name + "*") :
-                    parent.Children(name);
+                var hit = dis.TryGetValue(name, out var info);
```

Comment:
By not calling tryGetBySuffixedName() anymore (and its associated smarts for resolving multiple choices with the longest hit - the `Aggregate()`) you're reintroducing the bug fixed in R5 here: https://github.com/FirelyTeam/firely-net-sdk/pull/1859.

(I noticed Marten has not added this issue to a project - so this fix never made it into the readme for the august release).